SerialPrograms/Source/PokemonHome/Programs/PokemonHome_NationalDexSorter.cpp

```diff
@@ -0,0 +1,1532 @@
+    const uint lineCol1 =0xf44336;          // the red to check for in the name bar
+    const uint lineCol2 =0xf44336;          // the red to check for in the name bar
+    int eCount = 0;                         // if at any point you make too many moves before getting to the next step give up on the pokémon
+    const int eCountMax = 7;               // this is the max steps I allow before giving up on a given task
+
+PokemonLA::ItemCompatibility detect_item_compatibility(const ImageViewRGB32& screen,const ImageFloatBox& box,const uint& col1,const uint& col2,const int& superCheck){
+    // This is copied from the SkipToFullMoon code, I use it as a solid color check.
+    // I use it in various ways in order to reach my goals, not all of them make sense, but all work.
+
+    const bool replace_background = true;
+    ImageRGB32 region = filter_rgb32_range(
+        extract_box_reference(screen, box),
+        col1, col2, Color(0), replace_background
+    );
+
+    ImageStats stats = image_stats(region);
+    // std::cout << "Compability color " << stats.average.r << " " << stats.average.g << " " << stats.average.b << std::endl;
+    if (stats.average.r > stats.average.b + 50.0){
+        return PokemonLA::ItemCompatibility::COMPATIBLE;
+    }
+    if (stats.average.b > stats.average.r + 50.0){
+        return PokemonLA::ItemCompatibility::INCOMPATIBLE;
+    }
+    //I created these 2 checks for more specific colors, no idea how it works, just know it works :P
+    if (stats.average.b > 254.5 && stats.average.r > 252.5 && superCheck == 1){
+        return PokemonLA::ItemCompatibility::COMPATIBLE;
+    }
+    if (stats.average.r >  250 && superCheck == 2){
+        return PokemonLA::ItemCompatibility::COMPATIBLE;
+    }
+    return PokemonLA::ItemCompatibility::NONE;
+}
+
+int read_number(Logger& logger, const ImageViewRGB32& image2,const ImageFloatBox& box){
+    // This is the OCR read_number function, I only changed the log so that it says which BoxPage it's reading
+    ImageViewRGB32 image = extract_box_reference(image2, box);
+    std::string ocr_text = OCR::ocr_read(Language::English, image);
+    std::string normalized;
+    bool has_digit = false;
+    for (char ch : ocr_text){
+        //  4 is commonly misread as A.
+        if (ch == 'a' || ch == 'A'){
+            normalized += '4';
+            has_digit = true;
+        }
+        if ('0' <= ch && ch <= '9'){
+            normalized += ch;
+            has_digit = true;
+        }
+    }
+
+    if (!has_digit){
+        return -1;
+    }
+
+    int number = std::atoi(normalized.c_str());
+
+    std::string str;
+    for (char ch : ocr_text){
+        if (ch != '\r' && ch != '\n'){
+            str += ch;
+        }
+    }
+    logger.log("OCR Text: \"" + str + "\" -> \"" + normalized + "\" -> Box Page #" + std::to_string(number),COLOR_RED);
```

Comment:
you should call original function and add some more logs after this call. That way, you avoid duplicating the code

Reply:
I forgot to mention in the comment I also changed the code so that it accepts an ImageFloatBox, so that it only has to read a very very small part of the screen.
Now I figured I can send extract_box_reference as the ImageViewRGB32 argument instead of the whole screen. Done.